async_simple/coro/Lazy.h

```diff
@@ -393,6 +412,30 @@ class LazyBase {
     auto coAwaitTry() { return TryAwaiter(std::exchange(_coro, nullptr)); }
 
 protected:
+    template <isDerivedFromLazyLocal LazyLocal>
+    void setLazyLocal(std::unique_ptr<LazyLocal> base) {
+        logicAssert(this->_coro.operator bool(),
+                    "Lazy do not have a coroutine_handle "
+                    "Maybe the allocation failed or you're using a used Lazy");
+        auto& local = this->_coro.promise()._lazy_local;
+        if (local) {
+            delete local;
+        }
+        local = base.release();
+    }
+
+    template <isDerivedFromLazyLocal LazyLocal, typename... Args>
+    void setLazyLocal(Args&&... args) {
+        logicAssert(this->_coro.operator bool(),
+                    "Lazy do not have a coroutine_handle "
+                    "Maybe the allocation failed or you're using a used Lazy");
+        auto& local = this->_coro.promise()._lazy_local;
+        if (local) {
+            delete local;
+        }
+        local = new LazyLocal{std::forward<Args>(args)...};
+    }
```

Comment:
I prefer to implement it as:

```
template<class LazyLocalTy, class Args...>
Lazy<...> setLazyLocal(Args ...) {
     LazyLocalTy lazy_local_ty(args...);
     this->...->lazy_local = &lazy_local_ty;
     co_await *this;
}  
```

then we can manage its lifetime automatically.